spock-core/src/main/java/org/spockframework/runtime/SpockRuntime.java

```diff
@@ -77,6 +77,11 @@ public static void conditionFailedWithException(ErrorCollector errorCollector, @
         errorCollector.collectOrThrow(spockException); // this is our exception - it already has good message
         return;
       }
+      if (throwable instanceof MultipleFailuresError) {
```

Comment:
💬 I am not particularly happy about that, but I did not see a better solution.

Helpers are supposed to be used in then/expect blocks. The `ConditionRewriter` will wrap this call in a try-catch that will catch assertion errors thrown by helper methods and invoke this method. In case of the `@Verify` method, we will return in one of the earlier ifs. In case of the `@VerifyAll` method, it will throw a `MultipleFailuresError` that will bypass all checks and will be wrapped as a `ConditionFailedWithExceptionError`.

As an alternative, we can prevent calls to helper methods from being rewritten. It seemed like a further-reaching change to me, so I went with this option instead.

Reply:
Yes, makes sense. Implemented in [`34062b5` (#2112)](https://github.com/spockframework/spock/pull/2112/commits/34062b5d2b14a52167c9fc62131fe931b103b41f). I did not replace all usages of `MultipleFailuresError`, only those that are relevant to this change.